skada/datasets/_office_home.py

```diff
@@ -0,0 +1,338 @@
+        _download_office_home(
+            loader_spec.remote,
+            data_home,
+            data_home if loader_spec.extract_root else dataset_dir,
+        )
+
+    (X, y) = _load_office_home(
+        dataset_dir,
+        domain,
+        shuffle=shuffle,
+        random_state=random_state,
+    )
+
+    dataset = DomainAwareDataset()
+    dataset.add_domain(X, y, domain.value)
+
+    return (X, y) if return_X_y else dataset
+
+
+def _download_office_home(remote_spec: RemoteFileMetadata, download_dir, extract_dir):
+    _logger.info(f"Downloading OfficeHome from {remote_spec.url} to {download_dir}")
+    started_at = time.time()
+    dataset_path = _fetch_remote(remote_spec, dirname=download_dir)
+    finished_at = time.time() - started_at
+    _logger.info(f"Finished downloading in {finished_at:0.2f} seconds")
+
+    with zipfile.ZipFile(dataset_path, "r") as zip_ref:
+        zip_ref.extractall(extract_dir)
+
+    os.remove(dataset_path)
+
+
+def _load_office_home(
+    dataset_dir: Union[os.PathLike, str],
+    domain: OfficeHomeDomain,
+    shuffle: bool = False,
+    random_state: Union[None, int, np.random.RandomState] = None,
+) -> Bunch:
+    fullpath = Path(dataset_dir)
+    csv_path = str(fullpath) + "/" + "OfficeHomeResnet50.csv"
+
+    (X, y, domains) = parse_office_home_csv(csv_path)
+
+    # To select only one domain
+    domain_mask = np.char.lower(domains) == domain.value
+    X = X[domain_mask]
+    y = y[domain_mask]
+
+    if shuffle:
+        X, y = shuffle_arrays(X, y, random_state=random_state)
+
+    return X, y
+
+
+def _load_office_home_all(
+    dataset_dir: Union[os.PathLike, str],
+    shuffle: bool = False,
+    random_state: Union[None, int, np.random.RandomState] = None,
+) -> Bunch:
+    fullpath = Path(dataset_dir)
+    csv_path = str(fullpath) + "/" + "OfficeHomeResnet50.csv"
+
+    (X, y, domains) = parse_office_home_csv(csv_path)
+
+    if shuffle:
+        X, y, domains = shuffle_arrays(X, y, domains, random_state=random_state)
+
+    return X, y, domains
+
+
+def parse_office_home_csv(csv_path):
+    # Initialize lists to store features, labels, and domains
+    features = []
+    labels = []
+    domains = []
+
+    with open(csv_path, newline="") as csvfile:
+        # Create a CSV reader object
+        reader = csv.reader(csvfile, quotechar='"')
+
+        # Skip the header row
+        next(reader)
+
+        # Iterate over each row in the CSV file
+        for row in reader:
+            # Append the feature (first column) to the features list
+            features.append(eval(row[0]))
+            # Append the label (second column) to the labels list
+            labels.append(row[1])
+            # Append the domain (third column) to the domains list
+            domains.append(row[2])
+
+    X = np.array(features)
+    y = np.array(labels)
+    domains = np.array(domains)
+
+    return (X, y, domains)
+
+
+def shuffle_arrays(*arrays, random_state=None):
+    if random_state is None:
```

Comment:
We can use `random_state = check_random_state(random_state)` from sklearn to be consistent.